clang/test/CXX/over/over.match/over.match.funcs/over.match.class.deduct/p2.cpp

```diff
@@ -85,3 +85,13 @@ int main() {
 
 
 }
+
+namespace deduceTemplatedConstructor{
```

Comment:
I apologize for asking you to do additional work here but can you add the rest of the examples from [over.match.best.general p2.13](http://eel.is/c++draft/over.match.best.general#2.13).

It looks like we get all the test correct: https://godbolt.org/z/5MPrErPrq

and it would be nice to have those tests to prevent any possible future regressions.